### PR TITLE
Add Benchmarks

### PR DESCRIPTION
**Pull request by jiguantong**
- [x] Add benchmark for pangolin
- [x] Add benchmark for pangoro
- [x] Update weight for pangolin
- [x] Update weight for pangoro
- [x] Re-benchmark all with `--steps 50 --repeat 20`

### Fix 
- [x] pallet_elections_phragmen https://github.com/darwinia-network/darwinia-common/pull/1347
```shell
Error: Input("failed to submit candidacy")
```
- [x] pallet_membership
```shell
panicked at 'assertion failed: <Prime<T, I>>::get().is_some() && T::MembershipChanged::get_prime().is_some()'
```
- [ ] pallet_bounties https://github.com/darwinia-network/darwinia-common/issues/1360
```shell
Error: Input("Premature")
```
- [x] to_parachain_backing
```shell
Error: Input("MessageRejectedByLaneVerifier")
```
- [x] pallet_bridge_grandpa merge https://github.com/darwinia-network/darwinia-messages-substrate/pull/166
```shell
  error[E0603]: trait `WeightInfo` is private
    --> /home/workspace/darwinia-git/darwinia-common/node/runtime/pangolin/src/weights/pallet_bridge_grandpa.rs:57:54
     |
  57 | impl<T: frame_system::Config> pallet_bridge_grandpa::WeightInfo for WeightInfo<T> {
     |                                                      ^^^^^^^^^^ private trait
     |
  note: the trait `WeightInfo` is defined here
    --> /root/.cargo/git/checkouts/darwinia-messages-substrate-229c8f9c9ba36c43/a7bf82f/modules/grandpa/src/lib.rs:39:5
     |
  39 | use crate::weights::WeightInfo;
```

### CODE REVIEW

File: node/runtime/pangolin/Cargo.toml [hunk 1]
```diff
@@ -263,14 +263,34 @@ runtime-benchmarks = [
...
+	"pallet-treasury/runtime-benchmarks",
+	"pallet-utility/runtime-benchmarks",
+	"pallet-vesting/runtime-benchmarks",
 	# darwinia-network
 	"darwinia-ethereum/runtime-benchmarks",
 	"darwinia-evm/runtime-benchmarks",
 	"from-substrate-issuing/runtime-benchmarks",
 	"pallet-fee-market/runtime-benchmarks",
 	"to-parachain-backing/runtime-benchmarks",
+	"drml-common-runtime/runtime-benchmarks",
```
Comment (AsceticBear) [inline node/runtime/pangolin/Cargo.toml:293]: Please sort this alphabetically as others do.

File: node/runtime/pangolin/src/weights/mod.rs [hunk 2]
```diff
@@ -0,0 +1,23 @@
+pub mod frame_system;
```
Comment (AsceticBear) [inline node/runtime/pangolin/src/weights/mod.rs:1]: Add License?

File: node/runtime/pangolin/src/weights/pallet_treasury_treasury.rs [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+
+#![cfg_attr(rustfmt, rustfmt_skip)]
+#![allow(unused_parens)]
+#![allow(unused_imports)]
+
+use frame_support::{traits::Get, weights::Weight};
+use sp_std::marker::PhantomData;
+
+/// Weight functions for `pallet_treasury`.
```
Comment (AsceticBear) [inline node/runtime/pangolin/src/weights/pallet_treasury_treasury.rs:55]: Can we rename `pallet_treasury_treasury.rs`, seems weird.

File: node/runtime/pangoro/Cargo.toml [hunk 4]
```diff
@@ -214,10 +214,20 @@ runtime-benchmarks = [
...
+	"pallet-im-online/runtime-benchmarks",
+	"pallet-timestamp/runtime-benchmarks",
+	"pallet-treasury/runtime-benchmarks",
+	"pallet-scheduler/runtime-benchmarks",
 	# darwinia-network
 	"darwinia-ethereum/runtime-benchmarks",
 	"darwinia-evm/runtime-benchmarks",
-	"to-substrate-backing/runtime-benchmarks",
+	"pallet-fee-market/runtime-benchmarks",
+	"drml-common-runtime/runtime-benchmarks",
```
Comment (AsceticBear) [inline node/runtime/pangoro/Cargo.toml:230]: Format issue

File: node/runtime/pangoro/src/weights/mod.rs [hunk 5]
```diff
@@ -0,0 +1,7 @@
+pub mod frame_system;
```
Comment (AsceticBear) [inline node/runtime/pangoro/src/weights/mod.rs:1]: License?

File: node/runtime/pangolin/src/weights/pallet_treasury_treasury.rs (hunk 3 above)
Comment (jiguantong) [reply node/runtime/pangolin/src/weights/pallet_treasury_treasury.rs:55]: This is automatically generated by the benchmark to distinguish between `KtonTreasury` and `Treasury`

File: node/runtime/pangolin/Cargo.toml (hunk 1 above)
Comment (jiguantong) [reply node/runtime/pangolin/Cargo.toml:293]: Fixed

File: node/runtime/pangolin/src/weights/mod.rs (hunk 2 above)
Comment (jiguantong) [reply node/runtime/pangolin/src/weights/mod.rs:1]: Fixed

File: node/runtime/pangoro/Cargo.toml (hunk 4 above)
Comment (jiguantong) [reply node/runtime/pangoro/Cargo.toml:230]: Fixed

File: node/runtime/pangoro/src/weights/mod.rs (hunk 5 above)
Comment (jiguantong) [reply node/runtime/pangoro/src/weights/mod.rs:1]: Fixed

File: node/runtime/pangolin/src/pallets/membership.rs [hunk 6]
```diff
@@ -16,6 +17,14 @@ impl ChangeMembers<AccountId> for MembershipChangedGroup {
...
 		EthereumRelay::change_members_sorted(incoming, outgoing, sorted_new);
 	}
+
+	fn set_prime(prime: Option<AccountId>) {
+		<Prime<Runtime, TechnicalCollective>>::set(prime);
+	}
+
+	fn get_prime() -> Option<AccountId> {
+		<Prime<Runtime, TechnicalCollective>>::get()
+	}
```
Comment (AsceticBear) [inline node/runtime/pangolin/src/pallets/membership.rs:27]: @AurevoirXavier I am not familiar with this, please take a review here.

**Review by boundless-forest — APPROVED** [review]
LGTM

**Review by aurexav — APPROVED** [review]

**Comment by aurexav** [comment]
Close paritytech/substrate#845.

**Pull request merged**